test/src/test/java/hudson/slaves/JNLPLauncherTest.java

```diff
@@ -236,6 +238,18 @@ private void launchJnlpAndVerify(Computer c, ArgumentListBuilder args) throws Ex
         assertTrue(c.isOffline());
     }
 
+    @Test
+    public void changeLauncher() throws Exception {
+        Computer c = addTestAgent(false);
+        var name = c.getName();
+        var node = c.getNode();
+        assertThat(c.isLaunchSupported(), is(false));
+        var nodeCopy = (Slave) Jenkins.XSTREAM2.fromXML(Jenkins.XSTREAM2.toXML(node));
+        nodeCopy.setLauncher(new SimpleCommandLauncher("true"));
+        Jenkins.get().getNodesObject().replaceNode(node, nodeCopy);
+        assertThat(Jenkins.get().getComputer(name).isLaunchSupported(), is(true));
```

Comment:
Without patch, fails here with `false` because the `SimpleCommandLauncher` is not applied to computer.